### DISCUSSION TITLE
建议kbe能支持c++扩展开发，而不改动或者只改动一点点kbe代码

### DISCUSSION THREAD

**Discussion by qq952670771** [kbengine/kbengine#1365] [Ideas]
如题：一个项目中正常来说，战斗都是最耗性能的地方，这个时候用c++写的话，就会提升很多性能，对于非战斗逻辑，就用Python写，这样既有脚本的开发效率，又免去了性能问题

比如说：可以在def文件中指定某个远程函数的调用层，如果不填或者填Python，就传到Python层处理，如果填的cpp就转到c++层（新加个c++的解决方案，里面的多个c++项目，一一对应多个app，这样新扩展的c++层，也可以当成项目资产的一部分）处理，

**Comment by kebiao** [comment]
你可以写python的pyd库， 用c写。 在脚本里面调用。

**Reply by qq952670771** [reply]
因为写战斗不可避免的要获取场景中其他的实体，如果直接用c写的话，不可避免要与其他实体交互，如果写pyd库的话，获取不到当前进程中的实体啊

**Reply by kebiao** [reply]
只不过是用c写脚本，应该可以获取到

**Comment by qhf566** [comment]
谁有大厂,近5年内开发完的,国内没有运营过的,成品开发完毕的手机游戏,联系我809996677
